starter_kits/Python3/hlt/entity.py

```diff
@@ -83,11 +85,25 @@ def stay_still(self):
     def _generate(player_id):
         """
         Creates an instance of a ship for a given player given the engine's input.
+        If an instance with the same ship.id has previously been generated, that instance will be returned.
         :param player_id: The id of the player who owns this ship
         :return: The ship id and ship object
         """
+        # Read game engine input
         ship_id, x_position, y_position, halite = map(int, read_input().split())
-        return ship_id, Ship(player_id, ship_id, Position(x_position, y_position), halite)
+
+        # Check storage to see if ship already exists
+        # If the ship exists, update its position and halite
+        if ship_id in Ship.__ships.keys():    
+            old_ship = Ship.__ships[id]
```

Comment:
Typo: `id` should be `ship_id`
```suggestion
            old_ship = Ship.__ships[ship_id]
```